dali/operators/geometry/affine_transforms/rotate_transform.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/geometry/affine_transforms/transform_base_op.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/core/geom/transform.h"
+
+namespace dali {
+
+DALI_SCHEMA(RotateTransform)
+  .DocStr(R"code(Produces a rotation affine transform matrix.
+
+If another transform matrix is passed as an input, the operator applies rotation to the matrix provided.
+
+The number of dimensions will be 2 or 3 depending on whether the argument ``axis`` is provided or not,
+respectively.
+
+.. note::
+    The output of this operator can be fed directly to the ``MT`` argument of ``CoordTransform`` operator.
+)code")
+  .AddArg(
+    "angle",
+    R"code(Angle, in degrees.
+
+For two-dimensional data, the rotation is counter-clockwise, assuming the top-left corner is
+at ``(0,0)``. For three-dimensional data, the ``angle`` is a positive rotation around the provided
+axis.)code",
+    DALI_FLOAT, true)
+  .AddOptionalArg<std::vector<float>>(
+    "axis",
+    R"code(Applies **only** to three-dimension and is the axis around which to rotate the image.
+
+The vector does not need to be normalized, but it must have a non-zero length.
+
+Reversing the vector is equivalent to changing the sign of ``angle``.)code",
+    nullptr, true)
+  .AddOptionalArg<std::vector<float>>(
+    "center",
+    R"code(The center of the rotation operation.
+
+If provided, the number of elements should match the number of dimensions.)code",
+    nullptr, true)
+  .NumInput(0, 1)
+  .NumOutput(1)
+  .AddParent("TransformAttr");
+
+template <typename T>
+T deg2rad(T degrees) {
+  constexpr T k = T(1) / T(180);
+  return k * degrees;
+}
+
+/**
+ * @brief Rotate transformation.
+ */
+class RotateTransformCPU
+    : public TransformBaseOp<CPUBackend, RotateTransformCPU> {
+ public:
+  using SupportedDims = dims<2, 3>;
+
+  explicit RotateTransformCPU(const OpSpec &spec) :
+      TransformBaseOp<CPUBackend, RotateTransformCPU>(spec),
+      has_angle_input_(spec.HasTensorArgument("angle")),
+      has_axis_(spec.HasArgument("axis")),
+      has_axis_input_(spec.HasTensorArgument("axis")),
```

Comment:
This seems to be a rerecording pattern.
Do you think that there is a value in making this a function:
```
spec.CheckArgumentPresence(has_axis_, has_axis_input, "asix");
```?